apps/aecore/src/aec_chain_state.erl

```diff
@@ -279,6 +282,14 @@ node_root_hash(#node{type = block , content = X}) -> aec_blocks:root_hash(X).
 node_target(#node{type = header, content = X}) -> aec_headers:target(X);
 node_target(#node{type = block , content = X}) -> aec_blocks:target(X).
 
+maybe_add_genesis_hash(#{genesis_block_hash := undefined} = State, Node) ->
+    case node_height(Node) =:= 0 of
+        true  -> State#{genesis_block_hash => hash(Node)};
```

Comment:
I admit I still need to get a full understanding of this PR, but it looks like this change (that is in commit ["Update genesis hash on header insert as well"](https://github.com/aeternity/epoch/pull/403/commits/6847de2852f162d342e8fe2a850915fae044914e)) makes it possible for a peer to induce this node to amend its understanding of what *the* genesis block is. That looks not correct.

(Related tickets are [PT-153184647](https://www.pivotaltracker.com/story/show/153184647) and [PT-153184755](https://www.pivotaltracker.com/story/show/153184755)).

Reply:
@lucafavatella posting a new genesis block [rejection is handled](https://github.com/aeternity/epoch/blob/3539d7b89b0eb37b3e7cf4cb754cdcc8d6cb7e2b/apps/aecore/src/aec_chain_state.erl#L458) before that